src/dos/drive_cache.cpp

```diff
@@ -678,7 +685,7 @@ DOS_Drive_Cache::CFileInfo* DOS_Drive_Cache::FindDirInfo(const char* path, char*
 //		bool errorcheck = false;
 		pos = strchr(start,CROSS_FILESPLIT);
 		if (pos) { safe_strncpy(dir,start,pos-start+1); /*errorcheck = true;*/ }
-		else	 { strcpy(dir,start); };
+		else	 { safe_strncpy(dir, start, sizeof(dir)); };
```

Comment:
Let's format these two lines, because damn - this is weird to read.

Reply:
Very hard to read; did a bit more cleanup of similar stuff.

Can't wait until we get our clang reformatter online; all of these tucked-away and unwrapped statements will become immediately obviously.

Reply:
Oh; I also turned on effc++, and it helped prevent the double-free and read-after-free cases in the classes where I added private copy and assignment operators.  These are caught at compile-time, and thankfully there is no code that attempts to copy or assign these objects types.

Reply:
Frankly, I am afraid of using clang formatter indiscriminately - it will make it harder to merge-in changes from upstream. On the other hand, there's not much activity in the upstream anyway.

For projects at work, I ended up scripting clang-format to make it point out/automatically fix formatting errors only in lines touched by the developer - but it worked nicely, because that project used Gerrit and every single commit had its own little review - for GitHub I will need to modify it somewhat, to match the workflow better.

As for effc++ - usage resulted in nice improvements and overall code is more robust now :)